src/main/kotlin/net/atos/zac/zoeken/model/zoekobject/DocumentZoekObject.kt

```diff
@@ -0,0 +1,171 @@
+import java.util.stream.Collectors
+
+@NoArgConstructor // required for Java bean inspection
+data class DocumentZoekObject(
+    @Field("id")
+    var uuid: String,
+
+    @Field
+    private var type: String,
+
+    @Field("informatieobject_identificatie")
+    var identificatie: String? = null,
+
+    @Field("informatieobject_titel")
+    var titel: String? = null,
+
+    @Field("informatieobject_beschrijving")
+    var beschrijving: String? = null,
+
+    @Field("informatieobject_zaaktypeUuid")
+    var zaaktypeUuid: String? = null,
+
+    @Field("informatieobject_zaaktypeIdentificatie")
+    var zaaktypeIdentificatie: String? = null,
+
+    @Field("informatieobject_zaaktypeOmschrijving")
+    var zaaktypeOmschrijving: String? = null,
+
+    @Field("informatieobject_zaakId")
+    var zaakIdentificatie: String? = null,
+
+    @Field("informatieobject_zaakUuid")
+    var zaakUuid: String? = null,
+
+    @Field("informatieobject_zaakAfgehandeld")
+    var isZaakAfgehandeld: Boolean = false,
+
+    @Field("informatieobject_zaakRelatie")
+    var zaakRelatie: String? = null,
+
+    @Field("informatieobject_creatiedatum")
+    var creatiedatum: Date? = null,
+
+    @Field("informatieobject_registratiedatum")
+    var registratiedatum: Date? = null,
+
+    @Field("informatieobject_ontvangstdatum")
+    var ontvangstdatum: Date? = null,
+
+    @Field("informatieobject_verzenddatum")
+    var verzenddatum: Date? = null,
+
+    @Field("informatieobject_ondertekeningDatum")
+    var ondertekeningDatum: Date? = null,
+
+    @Field("informatieobject_ondertekeningSoort")
+    var ondertekeningSoort: String? = null,
+
+    @Field("informatieobject_vertrouwelijkheidaanduiding")
+    var vertrouwelijkheidaanduiding: String? = null,
+
+    @Field("informatieobject_auteur")
+    var auteur: String? = null,
+
+    @Field("informatieobject_status")
+    private var status: String? = null,
+
+    @Field("informatieobject_formaat")
+    var formaat: String? = null,
+
+    @Field("informatieobject_versie")
+    var versie: Long = 0,
+
+    @Field("informatieobject_bestandsnaam")
+    var bestandsnaam: String? = null,
+
+    @Field("informatieobject_bestandsomvang")
+    var bestandsomvang: Long = 0,
+
+    @Field("informatieobject_documentType")
+    var documentType: String? = null,
+
+    @Field("informatieobject_inhoudUrl")
+    var inhoudUrl: String? = null,
+
+    @Field("informatieobject_vergrendeldDoorNaam")
+    var vergrendeldDoorNaam: String? = null,
+
+    @Field("informatieobject_vergrendeldDoorGebruikersnaam")
+    var vergrendeldDoorGebruikersnaam: String? = null,
+
+    @Field("informatieobject_indicaties")
+    private var indicaties: MutableList<String>? = null,
+
+    @Field("informatieobject_indicaties_sort")
+    private var indicatiesVolgorde: Long = 0
+) : ZoekObject {
+    override fun getObjectId() = uuid
+
+    override fun getType() = ZoekObjectType.valueOf(type)
+
+    fun setType(type: ZoekObjectType) {
```

Comment:
pull this into the interface?

Reply:
That's not possible I think? With an abstract class it would be but an interface?